### DISCUSSION TITLE
How can I (Why I can't) get the keys or values in App.Current.Resources?

### DISCUSSION THREAD

**Discussion by AndrewKeepCoding** [microsoft/microsoft-ui-xaml#8166] [Q&A] [answered]
These works:
```csharp
var color = App.Current.Resources["SystemAccentColor"];  // #FF0078D4
bool containsKey = App.Current.Resources.ContainsKey("SystemAccentColor");  // true
App.Current.Resources.TryGetValue("SystemAccentColor", out var value);  // #FF0078D4
bool keyContains = App.Current.Resources.Keys.Contains("SystemAccentColor"); // true
```
But these doesn't work:
```csharp
var keys = App.Current.Resources.Keys.ToList();  // empty
var values = App.Current.Resources.Values.ToList();  // empty
```
By the way, this returns 639 items but doesn't contain "SystemAccentColor", so I assume the collection is not complete.
```csharp
// Returns false
bool mergedContains = App.Current.Resources.MergedDictionaries
    .SelectMany(x => x.Keys)
    .ToList()
    .Contains("SystemAccentColor");
```

**Comment by DarranRowe** [comment]
I was looking around at the documentation and I did find a paragraph on [system resources](https://learn.microsoft.com/en-us/windows/uwp/xaml-platform/themeresource-markup-extension#system-resources). But to put it simply, system resources are not in the Xaml resource dictionary.
I would guess that there are special cases in the implementation of the IMap interface that looks up system values for the system resources. For example, Windows.UI.ViewManagement.UISettings.GetColorValue(UIColorType.Accent) could be used for SystemAccentColor.

**Comment by AndrewKeepCoding** [comment, marked as answer]
Yep. It seems to be that there are some special cases implemented. I just run into this behavior when I was asked if one can create a list of those resources. Thanks @DarranRowe!
